.github/workflows/BuildAndTest.yml

```diff
@@ -4,49 +4,49 @@ name: Build and Test
 
 # Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+    # Triggers the workflow on push or pull request events but only for the main branch
+    push:
+        branches: [main]
+    pull_request:
+        branches: [main]
 
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
+    # Allows you to run this workflow manually from the Actions tab
+    workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-        with:
-          submodules: 'true'
-          token: '${{ secrets.PAT }}'
-      
-      - name: Setup NodeJS 16
-        uses: actions/setup-node@v2
-        with:
-          node-version: 16.x
-          
-      - name: Show NodeJS version    
-        run: npm --version
-      
-      - name: Install dependencies
-        run: npm install
-
-      - name: Compile contracts
-        run: npx hardhat compile
-        
-      - name: Grab API key
-        run:
-          echo ALCHEMY_API_KEY='${{ secrets.ALCHEMY_API }}' >> secrets/alchemy.env
-      
-      - name: Run tests
-        run:
-          npx hardhat test
+    # This workflow contains a single job called "build"
+    build:
+        # The type of runner that the job will run on
+        runs-on: ubuntu-latest
 
+        # Steps represent a sequence of tasks that will be executed as part of the job
+        steps:
+            # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+            - uses: actions/checkout@v2
```

Comment:
Should these also be v3?

Reply:
yeah these are v3 after merging main back in